Merge mock-debug-extension with main

CowTipper989 opened the pull request:
Adding the first working iteration of the VSCode extension to main, as well as a readme.

CharlesAverill commented on .vscode/extensions.json:
Is the `.vscode` directory necessary? I thought those were local configuration files

CowTipper989 replied:
The .vscode directory is necessary. It defines the launch configuration for the extension, this is not something we will want to change per machine.

CowTipper989 replied:
I had also thought they were just local configs before but when we didn't include them things got a little broken, upon looking into them more, yes they are important to keep lol

CharlesAverill commented on CHANGELOG.md:
Should go inside of a folder for the extension

CowTipper989 replied:
I will move the name extension to it's own folder

CharlesAverill commented on MyCheatsheet.md:
This shouldn't be pushed to main, if you think it'll help with development later you can add it to the [wiki](https://github.com/utdscheld/name/wiki)

CowTipper989 replied:
Good idea, done

CharlesAverill commented on ReadMe.md:
Needs to go in a folder for the extension

CowTipper989 replied:
I will move the name extension to it's own folder

CharlesAverill commented on name-as/.artifacts/mips_test.out:
Why was this removed?

CowTipper989 replied:
I didn't realize it was important, I will readd it

CharlesAverill commented on package.json:
Needs to go in a folder for the extension

CowTipper989 replied:
I will move the name extension to it's own folder

CharlesAverill commented:
After running `npm run watch` I'm stuck with this message and no debug window:
```
 *  Executing task: npm run watch 


> vsname@0.0.1 watch
> npm run -S esbuild-base -- --sourcemap --sources-content=false --watch


> vsname@0.0.1 esbuild-base
> esbuild ./src/extension.ts --bundle --tsconfig=./tsconfig.json --external:vscode --format=cjs --platform=node --outfile=dist/extension.js --sourcemap --sources-content=false --watch

[watch] build finished, watching for changes...
```

CowTipper989 commented:
> After running `npm run watch` I'm stuck with this message and no debug window:
> 
> ```
>  *  Executing task: npm run watch 
> 
> 
> > vsname@0.0.1 watch
> > npm run -S esbuild-base -- --sourcemap --sources-content=false --watch
> 
> 
> > vsname@0.0.1 esbuild-base
> > esbuild ./src/extension.ts --bundle --tsconfig=./tsconfig.json --external:vscode --format=cjs --platform=node --outfile=dist/extension.js --sourcemap --sources-content=false --watch
> 
> [watch] build finished, watching for changes...
> ```

This is intended behavior.

CowTipper989 commented on MyCheatsheet.md:
```diff
@@ -0,0 +1,188 @@
+
+-----
+
+# Now for Svelte
```
It has been moved to a wiki page in the repo

CowTipper989 commented:
> Even when I started the extension debugger with F5 I had no extension running, is there any other way to start it up? @CowTipper989

Upon looking into this a little further, I believe their was an issue with the launch config in .vscode, I'll submit a new commit with a corrected launch.json and the other fixed before our meeting tonight

CowTipper989 commented:
The branch I am requesting to merge to main has been overhauled based on comments made earlier.
Please review again and alert me if anything else should be changed.

CowTipper989 commented:
Made a quick edit to remove merge conflicts with the gitignore and readme
- Added a full readme
- Changed gitignore to point to a folder within the sub directory of name-as since it has moved there

qwe-q commented:
NACK. This tampers with the git history by squashing changes. Let's merge from the branch where all the work was done
